src/bream/core.py

```diff
@@ -92,67 +92,18 @@ def _is_native_type(spec: TypeSpec) -> bool:
 
 
 @dataclasses.dataclass(frozen=True, slots=True)
-class NoEncoderAvailable:
-    value: object
-
-
-@dataclasses.dataclass(frozen=True, slots=True)
-class UnencodableDictKey:
-    value: object
-
-
-EncodeError = NoEncoderAvailable | UnencodableDictKey
-
-
-@dataclasses.dataclass(frozen=True, slots=True)
-class UnsupportedBreamSpec:
-    """The bream spec specified for deserialisation is unsupported."""
-
-    bream_spec: int
-
-
-@dataclasses.dataclass(frozen=True, slots=True)
-class NoDecoderAvailable:
-    """No decoder is available for the given type_label."""
-
-    type_label: TypeLabel
-
-
-@dataclasses.dataclass(frozen=True, slots=True)
-class UnsupportedCoderVersion:
+class UnsupportedCoderVersionError(Exception):
     """The version requested for deserialisation is not supported."""
 
     coder: Coder[Any]
     version_provided: int
```

Comment:
I am surprised that you can so easily mix dataclasses with inheritance. Perhaps make these fiends keyword-only, so it is clearer which init argument is the error message and which arguments go into constructing the dataclass.

Reply:
"fiends" lol

Reply:
Lol. I'll leave it in.

Reply:
4417bda